janusgraph-test/src/test/java/org/janusgraph/graphdb/tinkerpop/gremlin/server/handler/HttpHMACAuthenticationHandlerTest.java

```diff
@@ -0,0 +1,198 @@
+        final ChannelFuture cf = createMock(ChannelFuture.class);
+
+        expect(msg.getMethod()).andReturn(HttpMethod.POST);
+        expect(msg.headers()).andReturn(headers).anyTimes();
+        expect(headers.get("Authorization")).andReturn("bogus");
+        expect(ctx.writeAndFlush(eqHttpStatus(UNAUTHORIZED))).andReturn(cf);
+        expect(cf.addListener(ChannelFutureListener.CLOSE)).andReturn(null);
+        expect(msg.release()).andReturn(false);
+
+        HttpHMACAuthenticationHandler handler = new HttpHMACAuthenticationHandler(authenticator);
+        replayAll();
+        handler.channelRead(ctx,(Object) msg);
+        verifyAll();
+    }
+
+    @Test
+    public void testChannelReadBasicAuth() {
+        final ChannelHandlerContext ctx = createMock(ChannelHandlerContext.class);
+        final FullHttpRequest msg = createMock(FullHttpRequest.class);
+        final HttpHeaders headers = createMock(HttpHeaders.class);
+        final Authenticator authenticator = createMock(Authenticator.class);
+        final ChannelFuture cf = createMock(ChannelFuture.class);
+        final String encodedUserNameAndPass = Base64.getEncoder().encodeToString("user:pass".getBytes());
+        expect(msg.getMethod()).andReturn(HttpMethod.POST);
+        expect(msg.headers()).andReturn(headers).anyTimes();
+        expect(msg.getUri()).andReturn("/");
+        expect(headers.get(eq("Authorization"))).andReturn("Basic " + encodedUserNameAndPass);
+        expect(ctx.fireChannelRead(isA(FullHttpRequest.class))).andReturn(ctx);
+        try {
+            expect(authenticator.authenticate(isA(Map.class))).andReturn(new AuthenticatedUser("foo"));
+        } catch (Exception e) {}
+        HttpHMACAuthenticationHandler handler = new HttpHMACAuthenticationHandler(authenticator);
+        replayAll();
+        handler.channelRead(ctx,(Object) msg);
+        verifyAll();
+    }
+
+    @Test
+    public void testChannelReadGetAuthToken() {
+        final ChannelHandlerContext ctx = createMock(ChannelHandlerContext.class);
+        final FullHttpRequest msg = createMock(FullHttpRequest.class);
+        final HttpHeaders headers = createMock(HttpHeaders.class);
+        final Authenticator authenticator = createMock(Authenticator.class);
+        final ChannelFuture cf = createMock(ChannelFuture.class);
+        final String encodedUserNameAndPass = Base64.getEncoder().encodeToString("user:pass".getBytes());
+        final Capture<Map<String, String>> credMap = new Capture();
+        expect(msg.getMethod()).andReturn(HttpMethod.GET);
+        expect(msg.headers()).andReturn(headers).anyTimes();
+        expect(msg.getUri()).andReturn("/session");
+        expect(headers.get(eq("Authorization"))).andReturn("Basic " + encodedUserNameAndPass);
+
+        try {
+            expect(authenticator.authenticate(and(isA(Map.class), capture(credMap)))).andReturn(new AuthenticatedUser("foo"));
+        } catch (Exception e) {}
```

Comment:
This is masking all sorts of possible exceptions, quietly.

Why not just annotate the method as being allowed to throw exceptions (but it shouldn't in practice, right?), and drop the try/catch clause? Similarly elsewhere.

Reply:
Absolutely. I meant to change this before I committed, but it's been sitting around for a while and I forgot.